Sources/TuistGenerator/Generator/SchemeDescriptorsGenerator.swift

```diff
@@ -735,9 +735,9 @@ final class SchemeDescriptorsGenerator: SchemeDescriptorsGenerating {
     /// - Parameters:
     ///     - environments: environment variables
     /// - Returns: XCScheme.EnvironmentVariable.
-    private func environmentVariables(_ environments: [String: String]) -> [XCScheme.EnvironmentVariable] {
-        environments.map { key, value in
-            XCScheme.EnvironmentVariable(variable: key, value: value, enabled: true)
+    private func environmentVariables(_ environments: [EnvironmentVariable]) -> [XCScheme.EnvironmentVariable] {
+        environments.map { environment in
+            XCScheme.EnvironmentVariable(variable: environment.key, value: environment.value, enabled: environment.isEnabled)
         }.sorted { $0.variable < $1.variable }
```

Comment:
I suspect we no longer need to sort as we'll now have a stable order as we'll persist the user defined order.